src/Post/Post.php

```diff
@@ -104,6 +104,36 @@ public static function boot()
         static::addGlobalScope(new RegisteredTypesScope);
     }
 
+    /**
+     * We override this so that we can use custom logic to.
+     */
+    protected function insertAndSetId(Builder $query, $attributes)
+    {
+        $attrsNoNumber = array_filter($attributes, function ($key) {
+            return $key !== 'number';
+        }, ARRAY_FILTER_USE_KEY);
+
+        $literalsSubquery = Post::query();
+
+        foreach ($attrsNoNumber as $attr => $value) {
+            //! DANGER!!! Literal Injection Vuln!!!
+            $literalsSubquery->selectRaw("'$value' as $attr");
+        }
+
+        $literalsSubquery->selectSub(Post::query()->where('discussion_id', $this->discussion_id)->selectRaw('count(*)+1'), 'number');
```

Comment:
```suggestion
        $literalsSubquery->selectSub(Post::query()->where('discussion_id', $this->discussion_id)->selectRaw('count(*)+1'), 'number');
```

Instead of `*` always use a column, that's much faster. Preferably one that is indexed. In this case this might work:

```php
$key = $query->getModel()->getKeyName();
```

These might potentially be an array, be cautious 🙈 Compound keys aren't native Laravel but a package exists for it.